src/services/git/Git.ts

```diff
@@ -58,17 +123,50 @@ export class Git {
     });
   }
 
-  getTextFileContent(path: string): Promise<string> {
-    return this.getFile(path).then((f) => {
-      if (!f) {
-        throw ErrorFactory.newInternalError(`Could not get content of ${path}`);
+  private getRepoContent(path: string) {
+    const params = GitHubUrlParser.getOwnerAndRepoName(this.repository.url);
+    const key = `${params.owner}:${params.repoName}:content:${path}`;
+
+    return this.cache.getOrSet(key, async () => {
+      try {
+        const result = await this.gitHubClient.getRepoContent(params.owner, params.repoName, path);
+        return result.data;
+      } catch (e) {
+        if (e.name !== 'HttpError' || e.status !== 404) {
+          throw e;
+        }
+
+        return null;
       }
-      return Buffer.from(f.data, 'base64').toString('utf-8');
     });
   }
 
-  private contentCacheKey(path: string): string {
-    const params = GitHubUrlParser.getOwnerAndRepoName(this.repository.url);
-    return `${params.owner}:${params.repoName}:content:${path}`;
+  private async followSymLinks(path: string, directory?: string): Promise<string> {
```

Comment:
Do we really want to use this method? See the comment above

Reply:
ditto